### PR TITLE
278 resume upload

### LINKED ISSUE #278: [UserMenteeApplication] Resume upload
Similar to https://github.com/agency-of-learning/PairApp/issues/179 but instead adding the resume upload to this form.

### PR DESCRIPTION
**Pull request by afogel**
## What's the change?

Add in functionality for applicant to upload resume. Improve UX of top section of show action, making sure that action buttons are shown only when authorized. Fix specs so that they're correctly testing empty state.
also, DS_Store continued to get tracked by git. I make sure to ignore it now.

## What key workflows are impacted?
Applicant must upload a resume when submitting their 

#### Admin view of show page
Screen Shot 2023-09-05 at 2 46 14 PM

#### Applicant view of show page
Screen Shot 2023-09-05 at 3 08 15 PM
Screen Shot 2023-09-05 at 3 10 50 PM

## Issue ticket number and link
Closes #278 

## Checklist before requesting a review

Please delete items that are not relevant.

- [X] Did you add appropriate automated tests?
- [X] Did you consider risks around security, performance, etc.?
- [X] Have you thought of misfiring code? e.g. too many loops, n+1, or how to handle nils?
- [X] Any validations to data needed?
- [X] Related to readability and consistency: Did you add comments and/or documentation? (README, Notion, etc.)
- [X] Is everything elevated to the highest level? (e.g., can logic be moved out of view into controllers, or out of controllers into models?)
- [X] Is there any linting that needs to be executed?
- [X] Did you leave helpful inline PR comments for the reviewer(s)?
- [X] Did you include instructions for how to test in the ticket?
- [X] Did you add any relevant tags and decide if this PR is a Draft vs. Ready for Review?
- [X] Do you have a plan for how this change should be rolled back? (e.g., how do you deal with a migration rollback? Is your feature released behind a feature flag?)

### CODE REVIEW

File: app/components/user_mentee_application/action_button_builder_component.rb [hunk 1]
```diff
@@ -1,31 +1,25 @@
...
-
-  def initialize(mentee_application:)
+  def initialize(mentee_application:, current_user:)
     @mentee_application = mentee_application
+    @current_user = current_user
   end
 
+  private
+
+  attr_reader :mentee_application, :current_user
```
Comment (afogel) [inline app/components/user_mentee_application/action_button_builder_component.rb:11]: need current user to authorize that buttons can/should be rendered.

File: app/components/user_mentee_application/section_component.html.erb [hunk 2]
```diff
@@ -1,6 +1,6 @@
 <section class='flex flex-col'>
-  <h1 class='font-bold text-2xl text-left mb-3 text-black-300 sm:text-center'><%= header_text %></h1>
+  <h2 class='text-2xl text-left tracking-tight mb-6 text-black-300'><%= header_text %></h2>
```
Comment (afogel) [inline app/components/user_mentee_application/section_component.html.erb:2]: not the header, should be h2.

File: app/views/user_mentee_applications/index.html.erb [hunk 3]
```diff
@@ -3,8 +3,8 @@
 <div class="p-2">
   <div class="tabs">
     <%= link_to 'In Review', "#", class:"tab tab-lifted tab-active" %>
-    <%= link_to 'Accepted', "#", class:"tab tab-lifted" %>
-    <%= link_to 'Denied', "#", class:"tab tab-lifted" %>
+    <%= link_to 'Accepted', "#", class:"tab tab-lifted tab-disabled" %>
```
Comment (afogel) [inline app/views/user_mentee_applications/index.html.erb:6]: these don't function, they should be disabled to reflect that.

File: app/views/user_mentee_applications/new.html.erb [hunk 4]
```diff
@@ -1,33 +1,33 @@
...
-  Welcome <%= current_user.first_name %>!
+<h1 class="text-center font-bold text-3xl my-8">
+  Welcome, <%= current_user.first_name %>!
 </h1>
 
 <%= form_with model: @user_mentee_application, local: true, html: { class: 'max-w-prose mx-auto' } do |form| %>
   <div class='flex flex-col ml-5 mr-5'>
     <%= render UserMenteeApplication::SectionComponent.new(header_text: 'Tell us about your journey') do %>
       <div class="flex flex-col mb-5">
-        <%= form.label :learned_to_code, 'How did you learn to code?', class: 'font-bold' %>
```
Comment (afogel) [inline app/views/user_mentee_applications/new.html.erb:9]: All text on the page was bold, which nullified the efficacy of using bold. Removed font-bold on labels to establish better visual hierarchy.

File: app/views/user_mentee_applications/show.html.erb [hunk 5]
```diff
@@ -1,70 +1,94 @@
-<div class="max-w-prose mx-auto p-4">
-  <section class="flex flex-col mb-3 mb-5 ">
-    <div class="mx-auto text-2xl text-red-500 mb-3">
-      <%= @user_mentee_application.current_status.to_s.humanize %>
+<div class="mx-auto p-4 my-8 max-w-prose">
+  <section class="flex flex-col my-4 mx-auto rounded-md border-2">
+    <div class="inline my-4 mx-auto pt-4">
+      <span class="text-md text-base-content/70">Application Status</span>
```
Comment (afogel) [inline app/views/user_mentee_applications/show.html.erb:4]: Status needs a label, it's not data that is self-explanatory enough for an applicant that the data can be freestanding.

File: app/views/user_mentee_applications/show.html.erb [hunk 6]
```diff
@@ -1,70 +1,94 @@
...
+      </span>
     </div>
-
-    <ul class="menu menu-horizontal bg-base-200">
-      <li> <%= render(UserMenteeApplication::ActionButtonBuilderComponent.new(mentee_application: @user_mentee_application)) %> </li>
-    </ul>
+    <span class="pb-4 inline mx-auto">
+      <% if @user_mentee_application.user == current_user %>
+        <p class="text-md p-6">
+          Thank you for submitting your application to join us at the <span class="font-semibold">Agency of Learning</span>!<br /><br />
```
Comment (afogel) [inline app/views/user_mentee_applications/show.html.erb:12]: Need copy to communicate expectations and the status of the application to applicants.

File: spec/components/user_mentee_application/action_button_builder_component_spec.rb [hunk 7]
```diff
@@ -3,49 +3,63 @@
 require 'rails_helper'
 
 RSpec.describe UserMenteeApplication::ActionButtonBuilderComponent, type: :component do
-  let!(:mentee_application) { create(:user_mentee_application) }
+  context 'when the current user is an admin' do
```
Comment (afogel) [inline spec/components/user_mentee_application/action_button_builder_component_spec.rb:6]: need to enforce auth on this component.

File: spec/components/user_mentee_application/action_button_builder_component_spec.rb [hunk 8]
```diff
@@ -3,49 +3,63 @@
...
+    context 'when current mentee application has an accepted status' do
+      it 'renders no text' do
+        mentee_application.mentee_application_states.last.accepted!
 
-      component = described_class.new(mentee_application:)
-      render_inline(component)
+        component = described_class.new(mentee_application:, current_user:)
+        render_inline(component)
 
-      expect(page).to have_text('')
```
Comment (afogel) [inline spec/components/user_mentee_application/action_button_builder_component_spec.rb:37]: this passes both when there are and are not buttons, therefore it is not the correct assertion. Has been updated.

File: spec/components/user_mentee_application/action_button_builder_component_spec.rb (hunk 7 above)
Comment (afogel) [reply spec/components/user_mentee_application/action_button_builder_component_spec.rb:6]: prolly should do a headless policy here...thoughts?

**Review by toyhammered — COMMENTED** [review]
Just want to leave a comment that this is going to conflict with some of the changes we are doing with the application status stuff because we are going to be adding notes. 

Can we change this to purely be for the resume upload only?

cc @JoshDevHub and @GALTdea cause you are working on the updates to the index/show page. (if you 2 want, you can decide if you want to merge this in and then just implement your changes ontop of it?)

**Pull request closed without merging**